### PR TITLE
enable core dumps for WSLGd and its children

### PR DESCRIPTION
**Pull request by hideyukn88**
enable core dumps for WSLGd and its children.

### CODE REVIEW

File: WSLGd/main.cpp [hunk 1]
```diff
@@ -186,6 +191,37 @@ try {
...
+    // create folder to store core files.
+    {
+        std::filesystem::create_directories(c_coreDir);
+        THROW_LAST_ERROR_IF(chmod(c_coreDir, 0777) < 0);
+    }
+
+    // update core_pattern.
+    {
+        wil::unique_file file_corePatternFile(fopen(c_corePatternFile, "w"));
+        THROW_LAST_ERROR_IF(!file_corePatternFile.get());
```
Comment (spronovo) [inline WSLGd/main.cpp:211]: Maybe don't make this error fatal just in case?

File: WSLGd/main.cpp [hunk 2]
```diff
@@ -186,6 +191,37 @@ try {
 
     SetupOptionalEnv();
 
+    // "ulimits -c unlimited" for core dumps.
+    {
```
Comment (benhillis) [inline WSLGd/main.cpp:195]: I don't think these scopes are buying you anything, no smart pointers inside going out of scope, can you clarify your intent?

**Review by benhillis — CHANGES_REQUESTED** [review]

File: WSLGd/main.cpp (hunk 2 above)
Comment (hideyukn88) [reply WSLGd/main.cpp:195]: Right, I was thinking to enable this path only when certain environment set, thus bracket was left over from "if (getenv("WSL2_WSLG_ENABLE_CORE_DUMP")" or I might add that eventually.

File: WSLGd/main.cpp (hunk 1 above)
Comment (hideyukn88) [reply WSLGd/main.cpp:211]: ok, I can change that.

File: WSLGd/main.cpp [hunk 3]
```diff
@@ -186,6 +191,37 @@ try {
...
+
+    // create folder to store core files.
+    {
+        std::filesystem::create_directories(c_coreDir);
+        THROW_LAST_ERROR_IF(chmod(c_coreDir, 0777) < 0);
+    }
+
+    // update core_pattern.
+    {
+        wil::unique_file file_corePatternFile(fopen(c_corePatternFile, "w"));
```
[the remark this answers is not in the document]
Comment (hideyukn88) [reply WSLGd/main.cpp:210]: fixed.

**Review by viniciusjarina — APPROVED** [review]

File: WSLGd/main.cpp [hunk 4]
```diff
@@ -186,6 +191,36 @@ try {
...
     SetupOptionalEnv();
 
+    // "ulimits -c unlimited" for core dumps.
+    struct rlimit limit;
+    limit.rlim_cur = RLIM_INFINITY;
+    limit.rlim_max = RLIM_INFINITY;
+    THROW_LAST_ERROR_IF(setrlimit(RLIMIT_CORE, &limit) < 0);
+
+    // create folder to store core files.
+    {
```
Comment (benhillis) [inline WSLGd/main.cpp:201]: scope not needed here either

**Review by benhillis — APPROVED** [review]

File: WSLGd/main.cpp (hunk 4 above)
Comment (hideyukn88) [reply WSLGd/main.cpp]: removed.

**Review by spronovo — APPROVED** [review]

**Pull request merged**